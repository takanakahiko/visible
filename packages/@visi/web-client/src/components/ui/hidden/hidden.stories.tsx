import React from 'react';

import { Hidden } from '.';

export default {
  title: 'Hidden',
  component: Hidden,
};

export const Normal = () => (
  <Hidden>This is only visible for screen readers</Hidden>
);
