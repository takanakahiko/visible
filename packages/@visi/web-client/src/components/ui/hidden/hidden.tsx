import React from 'react';

export interface HiddenProps {
  children: React.ReactNode;
}

export const Hidden = (props: HiddenProps) => {
  const { children } = props;
  return <div className="sr-only">{children}</div>;
};
