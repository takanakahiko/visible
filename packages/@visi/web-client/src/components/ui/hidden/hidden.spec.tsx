import { render } from '@testing-library/react';
import React from 'react';

import { Normal } from './hidden.stories';

describe('Hidden', () => {
  it('matches snapshot', () => {
    const { container } = render(<Normal />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
