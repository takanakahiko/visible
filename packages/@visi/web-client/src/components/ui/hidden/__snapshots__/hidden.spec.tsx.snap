// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hidden matches snapshot 1`] = `
<div
  class="sr-only"
>
  This is only visible for screen readers
</div>
`;
